src/main/java/org/wildfly/security/http/impl/ClientCertAuthenticationMechanism.java

```diff
@@ -95,4 +143,43 @@ public void evaluateRequest(HttpServerRequest request) throws HttpAuthentication
         }
     }
 
+    private boolean attemptReAuthentication(HttpServerRequest request) throws HttpAuthenticationException {
+        CachedIdentityAuthorizeCallback authorizeCallback = new CachedIdentityAuthorizeCallback(createIdentityCache(request), true);
+        try {
+            MechanismUtil.handleCallbacks(CLIENT_CERT_NAME, callbackHandler, authorizeCallback);
+        } catch (AuthenticationMechanismException e) {
+            throw e.toHttpAuthenticationException();
+        } catch (UnsupportedCallbackException e) {
+            throw log.mechCallbackHandlerFailedForUnknownReason(CLIENT_CERT_NAME, e).toHttpAuthenticationException();
+        }
+        if (authorizeCallback.isAuthorized()) {
+            return succeed(request);
+        }
+        return false;
+    }
+
+    private IdentityCache createIdentityCache(HttpServerRequest request) {
+        SSLSession sslSession = request.getSSLSession();
+        if (sslSession == null) {
+            return null;
+        }
+        return new IdentityCache() {
+            @Override
+            public void put(SecurityIdentity identity) {
+                sslSession.putValue(CACHED_IDENTITY_KEY, new CachedIdentity(getMechanismName(), identity));
```

Comment:
We had some previous work to ensure different mechanism instances backed by different security domains were not storing the SecurityIdentity in a globally accessible way - think we need to make sure that is still covered here.